awx/ui_next/src/components/Schedule/shared/DateTimePicker.jsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import { t } from '@lingui/macro';
+import { useField } from 'formik';
+import {
+  DatePicker,
+  isValidDate,
+  yyyyMMddFormat,
+  TimePicker,
+  FormGroup,
+} from '@patternfly/react-core';
+import styled from 'styled-components';
+import { required, validateTime, combine } from '../../../util/validators';
+
+const DateTimeGroup = styled.span`
+  display: flex;
+`;
+function DateTimePicker({ dateFieldName, timeFieldName, label }) {
+  const [dateField, dateMeta, dateHelpers] = useField({
+    name: `${dateFieldName}`,
+    validate: combine([required(null), isValidDate]),
+  });
+  const [timeField, timeMeta, timeHelpers] = useField({
+    name: `${timeFieldName}`,
+    validate: combine([required(null), validateTime()]),
+  });
+
+  const onDateChange = (inputDate, newDate) => {
+    dateHelpers.setTouched();
+    if (isValidDate(newDate) && inputDate === yyyyMMddFormat(newDate)) {
+      dateHelpers.setValue(new Date(newDate).toISOString().split('T')[0]);
+    }
+  };
+
+  return (
+    <FormGroup
+      fieldId={`schedule-${label}-datetime`}
+      helperTextInvalid={dateMeta.error || timeMeta.error}
+      isRequired
+      validated={
+        (!dateMeta.touched || !dateMeta.error) &&
+        (!timeMeta.touched || !timeMeta.error)
+          ? 'default'
+          : 'error'
+      }
+      label={`${label} date/time`}
+    >
+      <DateTimeGroup>
+        <DatePicker
+          aria-label={t`${label} date`}
+          {...dateField}
+          value={dateField.value.split('T')[0]}
+          onChange={onDateChange}
+        />
+        <TimePicker
```

Comment:
I think the default time picker placeholder is a bit confusing since it doesn't actually match up character for character with the format.  I wonder if we change it to something like `hh:mm AM/PM` if that would help folks know what the format is (if they're typing it in).  This probably isn't a huge deal for us since we always have a default value set but I found this confusing the first time I played with this component.  I thought it was expecting time in a 24 hour clock syntax like `21:45` instead of `09:45 PM`.